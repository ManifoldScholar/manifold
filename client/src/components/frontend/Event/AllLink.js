import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router-dom';
import lh from 'helpers/linkHandler';

export default class EventAllLink extends Component {

  static displayName = "Event.AllLink";

  static propTypes = {
    count: PropTypes.number,
    threshold: PropTypes.number,
    projectId: PropTypes.string
  };

  getAllLink() {
    let allLink = null;

    if (this.props.count > this.props.threshold) {
      allLink = (
        <div className="section-heading-utility-right">
          <Link
            to={lh.link("frontendProjectEvents", this.props.projectId)}
            className="button-primary"
          >
            See all Activity
          </Link>
        </div>
      );
    }

    return allLink;
  }

  render() {
    return this.getAllLink();
  }
}
