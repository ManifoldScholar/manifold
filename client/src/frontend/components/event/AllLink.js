import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import lh from "helpers/linkHandler";
import classNames from "classnames";

export default class EventAllLink extends Component {
  static displayName = "Event.AllLink";

  static propTypes = {
    count: PropTypes.number,
    threshold: PropTypes.number,
    project: PropTypes.object.isRequired,
    wrapperClasses: PropTypes.string
  };

  render() {
    const { project, wrapperClasses } = this.props;
    if (this.props.count <= this.props.threshold) return null;
    return (
      <div className={classNames(wrapperClasses)}>
        <Link
          to={lh.link("frontendProjectEvents", project.attributes.slug)}
          className="button-primary"
        >
          See all Activity
        </Link>
      </div>
    );
  }
}
