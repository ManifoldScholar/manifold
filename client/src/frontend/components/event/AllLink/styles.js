import styled from "@emotion/styled";

export const Wrapper = styled.div`
  align-self: center;
`;
