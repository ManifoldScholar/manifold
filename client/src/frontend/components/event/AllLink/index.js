import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { useTranslation } from "react-i18next";
import lh from "helpers/linkHandler";
import * as Styled from "./styles";

export default function EventAllLink({ threshold, entity }) {
  const { t } = useTranslation();

  if (entity.attributes.eventCount <= threshold) return null;

  return (
    <Styled.Wrapper>
      <Link
        to={lh.link("frontendProjectEvents", entity.attributes.slug)}
        className="button-primary"
      >
        <span className="button-primary__text">
          {t("navigation.see_all_activity")}
        </span>
      </Link>
    </Styled.Wrapper>
  );
}

EventAllLink.displayName = "Event.AllLink";

EventAllLink.propTypes = {
  threshold: PropTypes.number.isRequired,
  entity: PropTypes.object.isRequired
};
