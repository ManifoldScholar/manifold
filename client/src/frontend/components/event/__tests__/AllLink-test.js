import AllLink from "../AllLink";

describe("frontend/components/event/AllLink", () => {
  def("project", () => factory("project"));
  def("root", () => <AllLink threshold={3} entity={$project} />);

  it("matches the snapshot", () => {
    expect(shallow($root)).toMatchSnapshot();
  });
});
