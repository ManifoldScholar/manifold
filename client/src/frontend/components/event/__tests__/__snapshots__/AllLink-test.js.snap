// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`frontend/components/event/AllLink matches the snapshot 1`] = `
<Wrapper>
  <Link
    className="button-primary"
    to="/projects/slug-AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/events"
  >
    <span
      className="button-primary__text"
    >
      navigation.see_all_activity
    </span>
  </Link>
</Wrapper>
`;
